libopae/plugins/xfpga/metrics/metrics_utils.c

```diff
@@ -293,40 +285,38 @@ fpga_result enum_powermgmt_metrics(fpga_metric_vector *vector,
 		return FPGA_INVALID_PARAM;
 	}
 
-	snprintf_s_ss(sysfs_path, sizeof(sysfs_path), "%s/%s", sysfspath, PWRMGMT);
-
-	dir = opendir(sysfs_path);
-	if (NULL == dir) {
-		FPGA_MSG("can't find dir %s ", strerror(errno));
+	int gres = glob(sysfspath, GLOB_NOSORT, NULL, &pglob);
+	if (gres) {
+		FPGA_ERR("Failed pattern match %s: %s", sysfspath, strerror(errno));
+		globfree(&pglob);
```

Comment:
Do we always call globfree on all errors? Are there error cases where pglob will be NULL or not freeable?
Is this the only place where we call globfree? Don't we still need it before we exit the function?

Reply:
Fixed